controllers/helpers.go

```diff
@@ -126,6 +129,50 @@ func validateLabelSelector(selector labels.Selector) error {
 	return nil
 }
 
+// transformCloudSpecToHostsSpec from a cloud spec disruption, get all ip ranges of services provided and transform them into a list of hosts spec
+func transformCloudSpecToHostsSpec(log *zap.SugaredLogger, cloudManager *cloudservice.CloudProviderManager, cloudSpec *v1beta1.NetworkDisruptionCloudSpec) []v1beta1.NetworkDisruptionHostSpec {
+	hosts := []v1beta1.NetworkDisruptionHostSpec{}
+	clouds := map[cloudtypes.CloudProviderName][]string{
+		cloudtypes.CloudProviderAWS:     cloudSpec.AWS,
+		cloudtypes.CloudProviderDatadog: cloudSpec.Datadog,
+		cloudtypes.CloudProviderGCP:     cloudSpec.GCP,
+	}
+
+	for cloudName, serviceList := range clouds {
+		var ips []string
+
+		var err error
+
+		if len(serviceList) == 0 {
+			ips, err = cloudManager.GetServiceIPRanges(cloudName, "")
+			if err != nil {
+				log.Errorf("could not retrieve the ip range of all services of %s", cloudName)
```

Comment:
We need to log the error message here and below. The only errors I can possibly see coming from this function are "cloud provider doesn't exist" and "service name doesn't exist", both of which should already have been caught via validation before we get to this point. So these might be serious errors

Reply:
Chose to log it in the createChaosPods func instead, wdyt?